src/renderers/webgl-fallback/nodes/GLSLNodeBuilder.js

```diff
@@ -1320,8 +1326,11 @@ void main() {
 
 		if ( this.material !== null ) {
 
-			this.vertexShader = this._getGLSLVertexCode( shadersData.vertex );
-			this.fragmentShader = this._getGLSLFragmentCode( shadersData.fragment );
+			const name = this.material.name;
+			const shaderName = ( name !== '' ) ? name : this.material.constructor.name;
```

Comment:
How about adding `name` to the `shadersData` object? In this way, there is no need to change the signature of  `_getGLSLVertexCode()` and `_getGLSLFragmentCode()`.

Reply:
Doesn't that require changing the signature?

Reply:
You're right. I've thought the entire `shadersData` object is passed to the methods. In this case, the code seems fine as it is.